src/main.rs

```diff
@@ -95,29 +95,13 @@ fn parse_args() -> anyhow::Result<MacheteArgs> {
     })
 }
 
-fn collect_paths(path: &Path, skip_target_dir: bool) -> Vec<PathBuf> {
-    // Find directory entries.
-    let walker = WalkDir::new(path).into_iter();
-
-    let manifest_path_entries = if skip_target_dir {
-        walker
-            .filter_entry(|entry| !entry.path().ends_with("target"))
-            .collect()
-    } else {
-        walker.collect::<Vec<_>>()
-    };
-
-    manifest_path_entries
-        .into_iter()
-        .filter_map(|entry| match entry {
-            Ok(entry) if entry.file_name() == "Cargo.toml" => Some(entry.into_path()),
-            Err(err) => {
-                eprintln!("error when walking over subdirectories: {}", err);
-                None
-            }
-            _ => None,
-        })
-        .collect()
+fn build_toml_file_iterator(path: &Path, args: &MacheteArgs) -> impl Iterator<Item = Result<ignore::DirEntry, ignore::Error>> {
+    let mut walk_builder = ignore::WalkBuilder::new(path);
+    walk_builder.filter_entry(|entry| entry.file_name() == "Cargo.toml");
+    if args.skip_target_dir {
+        walk_builder.add_ignore("target/");
```

Comment:
I see this can return an `Option<Error>`; should we propagate it to the caller?

```suggestion
        if let Some(error) = walk_builder.add_ignore("target/") {
            return Err(error);
        }
```